### PR TITLE
ref: Type check post process forwarder

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

**Review by evanh — APPROVED** [review]

**Review by untitaker — APPROVED** [review]

File: src/sentry/post_process_forwarder/post_process_forwarder.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def _build_streaming_consumer(
     ) -> StreamProcessor[KafkaPayload]:
         configure_metrics(MetricsWrapper(metrics.backend, name="eventstream"))
 
-        cluster_name = settings.KAFKA_TOPICS[topic]["cluster"]
+        cluster_name = settings.KAFKA_TOPICS[topic]["cluster"]  # type: ignore
```
Comment (asottile-sentry) [inline src/sentry/post_process_forwarder/post_process_forwarder.py:100]: we should be able to annotate `KAFKA_TOPICS` in the settings file and fix this ignore

File: tests/sentry/post_process_forwarder/test_post_process_forwarder.py [hunk 2]
```diff
@@ -45,10 +45,9 @@ def kafka_message_payload() -> Any:
 
 class PostProcessForwarderTest(TestCase):
     def _get_producer(self, cluster_name: str) -> Producer:
+        bootstrap_servers = settings.KAFKA_CLUSTERS[cluster_name]["common"]["bootstrap.servers"]  # type: ignore
```
Comment (asottile-sentry) [inline tests/sentry/post_process_forwarder/test_post_process_forwarder.py:48]: ""

**Review by untitaker — APPROVED** [review]

File: tests/sentry/post_process_forwarder/test_post_process_forwarder.py (hunk 2 above)
Comment (untitaker) [reply tests/sentry/post_process_forwarder/test_post_process_forwarder.py:48]: not sure what this comment means, maybe it got resolved?

you could use `get_topic_definition` here

**Review by asottile-sentry — APPROVED** [review]

File: tests/sentry/post_process_forwarder/test_post_process_forwarder.py (hunk 2 above)
Comment (lynnagara) [reply tests/sentry/post_process_forwarder/test_post_process_forwarder.py]: @untitaker do you know why this isn't resolving the right cluster?

Comment (lynnagara) [reply tests/sentry/post_process_forwarder/test_post_process_forwarder.py]: Oh nvm, i see i'm using it wrong

**Pull request merged**